### PR TITLE
build: ♻️ 👷 🔧 turn on `strictNullChecks`

### PR DESCRIPTION
**Pull request by Jameskmonger**
I apologise for the size of this PR 😓 

This PR enables `strictNullChecks` and fixes all `600`+ issues that were uncovered by that check. It also enables `strictBindCallApply`, but there weren't any errors from that.

It also converts two plugins to TypeScript (`rs:hans` and `rs:transform_command`).

I have tested the game and appears to still be working. I avoided changing functionality wherever possible, mostly these changes are to TypeScript type annotations only.

---

from this:
image

to this:
image

### CODE REVIEW

File: src/engine/action/hook/task.ts
Comment (nyxkrage) [inline src/engine/action/hook/task.ts]: We could probably use `this.task!.delay` since this path requires it to be defined to some not 0 value

File: src/engine/action/hook/task.ts
Comment (nyxkrage) [inline src/engine/action/hook/task.ts]: Same as with `delay` above

File: src/engine/action/pipe/item-swap.action.ts
Comment (nyxkrage) [inline src/engine/action/pipe/item-swap.action.ts:50]: We should probably add `// eslint-disable-next-line no-non-null-assertion` in these cases, where we know that the assertion is true as to not clutter up the warnings.

File: src/engine/config/item-config.ts
Comment (nyxkrage) [inline src/engine/config/item-config.ts:189]: Should `key` be able to be undefined here, that seems wrong

File: src/engine/config/npc-config.ts
Comment (nyxkrage) [inline src/engine/config/npc-config.ts:89]: Same here as in regards to the `translateItemConfig`

File: src/engine/plugins/loader.ts
Comment (nyxkrage) [inline src/engine/plugins/loader.ts]: Again here we can just disable the eslint check for the following line

File: src/engine/util/files.ts
Comment (nyxkrage) [inline src/engine/util/files.ts]: I believe it would default to `false` here if `useWhitelist` is passed as undefined. See the following snippet for a demonstration
```js
useWhitelist = undefined

console.log("refactor")
if(!(useWhitelist || true)) {
    console.log("blacklist")
} else {
    console.log("whitelist")
}

console.log("refactor (default false)")
if(!(useWhitelist || false)) {
    console.log("blacklist")
} else {
    console.log("whitelist")
}

console.log("original")
if(!useWhitelist) {
    console.log("blacklist")
} else {
    console.log("whitelist")
}
```

File: src/engine/util/queue.ts
Comment (nyxkrage) [inline src/engine/util/queue.ts]: Do we want to throw here of just return `TItem | undefined` like for example `Array.at` does?
In case we do want to throw, it should probably be documented that the method *CAN* throw.

File: src/engine/world/actor/actor.ts
Comment (nyxkrage) [inline src/engine/world/actor/actor.ts:258]: Casting to `unkown` and then to the type feels like a hack that should probably have some form of corresponding comment.

File: src/engine/world/actor/npc.ts
Comment (nyxkrage) [inline src/engine/world/actor/npc.ts:149]: Should there be a warning here, having a drop, but a 0 amount seems like it would be cause for either an error or a warning at least

File: src/engine/world/actor/walking-queue.ts
Comment (nyxkrage) [inline src/engine/world/actor/walking-queue.ts]: This should be an unreachable branch since if the queue length is 0, we a returning early

File: src/engine/world/actor/walking-queue.ts
Comment (nyxkrage) [inline src/engine/world/actor/walking-queue.ts]: We wont be in this block if queue is 0, so this is also unreachable

File: src/engine/world/instances.ts
Comment (nyxkrage) [inline src/engine/world/instances.ts:99]: This should be optional according to the docs

File: src/engine/world/map/collision-map.ts
Comment (nyxkrage) [inline src/engine/world/map/collision-map.ts]: No reason to keep the comment here

File: src/engine/world/skill-util/harvest-skill.ts
Comment (nyxkrage) [inline src/engine/world/skill-util/harvest-skill.ts]: Is there a reason this is marked as a TODO and isn't just called?

File: src/engine/world/skill-util/harvest-skill.ts
Comment (nyxkrage) [inline src/engine/world/skill-util/harvest-skill.ts]: Same as above

File: src/engine/world/world.ts
Comment (nyxkrage) [inline src/engine/world/world.ts]: I would assume we should probably just return an empty Array

**Review by nyxkrage — CHANGES_REQUESTED** [review]

File: src/engine/world/skill-util/harvest-skill.ts
Comment (Jameskmonger) [reply src/engine/world/skill-util/harvest-skill.ts]: my mistake, I didn't realise `playAnimation` could accept a `null`. I will change now

File: src/engine/config/item-config.ts
Comment (Jameskmonger) [reply src/engine/config/item-config.ts:189]: these always get deep merged when `key` is `undefined`, it's a way to handle "extended" items.

Comment (Jameskmonger) [reply src/engine/config/item-config.ts:189]: I think the API could probably use some work in future though just to make that clearer

File: src/engine/config/npc-config.ts
Comment (Jameskmonger) [reply src/engine/config/npc-config.ts:89]: This one is for extended NPCs, so when `npcKey` is `undefined` here, it will get merged with something else with a key.

File: src/engine/world/instances.ts
Comment (Jameskmonger) [reply src/engine/world/instances.ts:99]: doesn't need to be, and makes the API more complicated, so have removed

File: src/engine/world/world.ts
Comment (Jameskmonger) [reply src/engine/world/world.ts]: I have stopped the param being `null` entirely 😄

**Review by nyxkrage — APPROVED** [review]
This looks good to me now, but should probably have second pair of eyes on it too before merging.

**Review by Katilith — APPROVED** [review]
Thanks for getting this done!

**Pull request merged**
